src/Simplify.cpp

```diff
@@ -3908,6 +3924,68 @@ class Simplify : public IRMutator {
             IRMutator::visit(op);
         }
     }
+    template <typename T>
+    Expr hoist_slice_vector(Expr e) {
+        const T *op = e.as<T>();
+        internal_assert(op);
+
+        debug(4) << "Trying to hoist slice vector " << (Expr) op << "\n";
+
+        const Call *call_a = op->a.template as<Call>();
+        const Call *call_b = op->b.template as<Call>();
+
+        internal_assert(call_a && call_b &&
+                        call_a->is_intrinsic(Call::slice_vector) &&
+                        call_b->is_intrinsic(Call::slice_vector));
+
+        const int64_t *start_lane_a = as_const_int(call_a->args[1]);
+        const int64_t *start_lane_b = as_const_int(call_b->args[1]);
+        if (*start_lane_a != *start_lane_b) {
+            debug(4) << " ...unable to hoist - differing start lanes\n";
+            return e;
+        }
+
+        const int64_t *stride_a = as_const_int(call_a->args[2]);
+        const int64_t *stride_b = as_const_int(call_b->args[2]);
+        if (*stride_a != 1 || *stride_b != 1) {
```

Comment:
I think when I first wrote it i was trying to limit the scope, but I'll post a patch in which the strides need to be equal and thats all.